packages/shared/src/components/cards/CollectionCard/CollectionCard.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { forwardRef, Ref, useState } from 'react';
+import classNames from 'classnames';
+import { Container, PostCardProps } from '../common';
+import FeedItemContainer from '../FeedItemContainer';
+import { CollectionCardHeader } from './CollectionCardHeader';
+import {
+  getPostClassNames,
+  FreeformCardTitle,
+  CardSpace,
+  CardButton,
+} from '../Card';
+import { WelcomePostCardFooter } from '../WelcomePostCardFooter';
+import ActionButtons from '../ActionButtons';
+import PostMetadata from '../PostMetadata';
+
+export const CollectionCard = forwardRef(function CollectionCard(
+  {
+    children,
+    post,
+    domProps = {},
+    onUpvoteClick,
+    onCommentClick,
+    onMenuClick,
+    onShare,
+    onShareClick,
+    openNewTab,
+    onReadArticleClick,
+    onPostClick,
+  }: PostCardProps,
+  ref: Ref<HTMLElement>,
+) {
+  const [headingFull, toggleHeadingFull] = useState(false);
+  const clamp = (() => {
+    if (post.image) {
+      return 'line-clamp-3';
+    }
+
+    return post.contentHtml ? 'line-clamp-4' : 'line-clamp-9';
+  })();
```

Comment:
this is duplicated code, I'll extract into separate hook

Reply:
If its quicker as this now leave it, more important to cleanup other stuff. Up to you 👍